flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/stream/StreamingFileWriter.java

```diff
@@ -134,13 +143,19 @@ public void notifyCheckpointComplete(long checkpointId) throws Exception {
 
 	private void commitUpToCheckpoint(long checkpointId) throws Exception {
 		helper.commitUpToCheckpoint(checkpointId);
+
+		NavigableMap<Long, Set<String>> newPartitions = this.newPartitions.headMap(checkpointId, true);
+		Set<String> partitions = new HashSet<>(committablePartitions);
+		committablePartitions.clear();
+		newPartitions.values().forEach(partitions::addAll);
+		newPartitions.clear();
```

Comment:
Will clear the headMap also cleans the part before the current checkpoint in `this.newPartitions` ?

Reply:
Yes